File: back/src/main/java/com/openclassrooms/mddapi/common/ApiRoutes.java
package com.openclassrooms.mddapi.common;

public class ApiRoutes {

	public static final String BASE_URL = "/api";


	/* Auth URL */
	public static final String AUTH_URL = BASE_URL+"/auth";

	public static final String LOGIN_URL = AUTH_URL+"/login";

	public static final String REGISTER_URL = AUTH_URL+"/register";

	public static final String ME_URL = AUTH_URL+"/me";

	public static final String JWT_REFRESH_URL = AUTH_URL+"/refresh";

	/* Users URL */
	public static final String USER_URL = BASE_URL+"/user";

	public static final String USER_UPDATE_URL = USER_URL+"/update";

	public static final String USER_PASSWORD_URL = USER_URL+"/password";

	public static final String USER_SUBSCRIBE_TOPIC_URL = USER_URL+"/subscribe/{topicName}";

	public static final String USER_UNSUBSCRIBE_TOPIC_URL = USER_URL+"/unsubscribe/{topicName}";

	/* Topics URL */
	public static final String TOPIC_URL = BASE_URL+"/topics";

}

File: front/src/app/core/api/api-routes.ts
import {environment} from '../../../environments/environment';

export const ApiRoutes = {
  auth: {
    login: `${environment.apiBaseUrl}/auth/login`,
    register: `${environment.apiBaseUrl}/auth/register`,
    me: `${environment.apiBaseUrl}/auth/me`,
    refresh: `${environment.apiBaseUrl}/auth/refresh`,
  },
  user: {
    update: `${environment.apiBaseUrl}/user/update`,
    changePassword: `${environment.apiBaseUrl}/user/password`,
    subscribe: (topicName: string) => `${environment.apiBaseUrl}/user/subscribe/${topicName}`,
    unsubscribe: (topicName: string) => `${environment.apiBaseUrl}/user/unsubscribe/${topicName}`,
  },
  topics: `${environment.apiBaseUrl}/topics`,
  posts: {
    base: `${environment.apiBaseUrl}/posts`,
    feed: `${environment.apiBaseUrl}/posts/feed`,
    detail: (postId: number) => `${environment.apiBaseUrl}/posts/${postId}`,
    comment: (postId: number) => `${environment.apiBaseUrl}/posts/${postId}/comment`,
  }
}
